### PR TITLE
Impl From<NonZero<T>> for T

### PR DESCRIPTION
**Pull request by csmoe**
Closes https://github.com/rust-lang/rust/issues/54171

r? @SimonSapin

### CODE REVIEW

**Comment by SimonSapin** [comment]
Looks good to me. Let’s get team sign off since impls are insta-stable:

@rfcbot fcp merge

File: src/test/run-pass/nonzero.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+fn main() {
+    use std::num::NonZeroU32;
+    let nz = NonZeroU32::new(5).unwrap();
+    let num: u32 = nz.into();
```
Comment (scottmcm) [inline src/test/run-pass/nonzero.rs]: nit: perhaps just make this a doctest on the impl, so users can see it?

File: src/libcore/num/mod.rs [hunk 2]
```diff
@@ -93,6 +93,13 @@ assert_eq!(size_of::<Option<std::num::", stringify!($Ty), ">>(), size_of::<", st
 
             }
 
+            #[stable(feature = "nonzero", since = "1.28.0")]
```
Comment (scottmcm) [inline src/libcore/num/mod.rs]: nit: I think it'll be [1.31](https://forge.rust-lang.org/) now?

File: src/test/run-pass/nonzero.rs (hunk 1 above)
Comment (csmoe) [reply src/test/run-pass/nonzero.rs]: most of the types like `String`/`PathBuf` in std doesn't have specific docs for their `From` implementation, and invoking `.from/.into` is kind of intuition, so I think that's fine without doctest? but if it's really needed, I'd add it.

Comment (scottmcm) [reply src/test/run-pass/nonzero.rs]: Definitely not a _need_.  I just found a run-pass test odd for this, as `x.py test src/libcore` doesn't run those.  `src/libcore/tests/nonzero.rs` might also be a reasonable place for the test.

Comment (SimonSapin) [reply src/test/run-pass/nonzero.rs]: `src/libcore/tests/nonzero.rs` also makes sense to me.

**Comment by SimonSapin** [comment]
The CI failure means this should use a new `feature` name, something like `from_nonzero`.

**Pull request merged**